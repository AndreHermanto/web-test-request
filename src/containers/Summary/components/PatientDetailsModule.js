import React, { Component } from 'react';
import { 
  Row, 
  Col,
  Glyphicon,
  Button 
} from 'react-bootstrap';
import { 
  SummaryBox,
  SummaryHeading,
  SummaryBreakLine,
  SummaryTitle,
  SummaryDetails,
  SummaryNotes,
  SectionHeading
} from '../summaryStyled';
import { 
  FileList,
  FileLink
} from './../../../components/SharedStyle';
import { isoToShortDate } from './../../../components/dateConvert';

/**
* PatientDetailsModule summary - UI for summary page to display all patient details/clinical info data.
*/
class PatientDetailsModule extends Component
{
  constructor(props) {
    super(props);
    this.handleDetailsClick = this.handleDetailsClick.bind(this);
    this.handleClinicalInfoClick = this.handleClinicalInfoClick.bind(this);
  }

  handleDetailsClick() {
    this.props.handleOnClick('step2');
    return 2;
  }

  handleClinicalInfoClick() {
    this.props.handleOnClick('step3');
    return 3;
  }

  render() {
    let clinicalNotes = this.props.clinicalInfo.clinicalInfo.split(/\r?\n/);
    let relevantInvestigationNotes = this.props.clinicalInfo.relevantInvestigation.split(/\r?\n/);
    let familyNotes = this.props.clinicalInfo.familyHistory.split(/\r?\n/);
    let consanguinityNotes = this.props.clinicalInfo.consanguinityInfo.split(/\r?\n/);
    return(
    <SummaryBox>
      <SummaryHeading> Patient - {this.props.patientDetails.firstName + ' ' + this.props.patientDetails.lastName} </SummaryHeading>
      <SummaryBreakLine/>
      <Row>
        <Col md={12}>
          <SectionHeading> 
            Details
            <Button 
              bsStyle="link"
              onClick={this.handleDetailsClick}
            >
              <Glyphicon glyph="pencil"/> Edit
            </Button>
          </SectionHeading>
        </Col>
        <Col md={12}>
          <SummaryDetails> 
            <SummaryTitle> Date of Birth: </SummaryTitle>
            {isoToShortDate(this.props.patientDetails.dob)}
          </SummaryDetails>
        </Col>
        {
          this.props.patientDetails.medicalRecordNo !== '' &&
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Medical record number:  </SummaryTitle>
              {this.props.patientDetails.medicalRecordNo}
            </SummaryDetails>
          </Col>
        }
        <Col md={12}>
          <SummaryDetails> 
            <SummaryTitle> Gender:  </SummaryTitle>
              {this.props.patientDetails.gender === 'Other' ?
                this.props.patientDetails.genderOther + ' (Other)' :
                this.props.patientDetails.gender
              } 
          </SummaryDetails>
        </Col>
        {
          this.props.patientDetails.deceased !== false &&
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Deceased:  </SummaryTitle>
              {this.props.patientDetails.deceased ? 'Yes' : 'No'}
            </SummaryDetails>
          </Col>
        }
        {
          (this.props.patientDetails.deceased !== false && this.props.patientDetails.sampleSource !== '') &&
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Sample source:  </SummaryTitle>
              {this.props.patientDetails.sampleSource}
            </SummaryDetails>
          </Col>
        }
        <Col md={12}>
          <SectionHeading> 
            Clinical Information
            <Button 
              bsStyle="link"
              onClick={this.handleClinicalInfoClick}
            >
              <Glyphicon glyph="pencil"/> Edit
            </Button>
          </SectionHeading>
        </Col>
        <Col md={12}>
          <SummaryTitle> Clinical note </SummaryTitle>
          <SummaryNotes>
          {
            clinicalNotes.length > 1 ? 
            clinicalNotes.map((n, i) => {
              return <p key={i} style={{fontWeight:200}}>{n}</p>;
            }) 
            : this.props.clinicalInfo.clinicalInfo
          }
          </SummaryNotes>
        </Col>
        {
          this.props.clinicalInfo.relevantInvestigation !== '' &&
          <Col md={12}>
            <SummaryTitle> Relevant investigation </SummaryTitle>
            <SummaryNotes>
              {
                relevantInvestigationNotes.length > 1 ?
                relevantInvestigationNotes.map((ri, i) => {
                  return <p key={i} style={{fontWeight:200}}>{ri}</p>;
                })
                : this.props.clinicalInfo.relevantInvestigation
              }
            </SummaryNotes>
          </Col>
        }
        {
          this.props.clinicalInfo.familyHistory !== '' &&
          <Col md={12}>
            <SummaryTitle> Family history </SummaryTitle>
            <SummaryNotes>
              {
                familyNotes.length > 1 ?
                familyNotes.map((ri, i) => {
                  return <p key={i} style={{fontWeight:200}}>{ri}</p>;
                })
                : this.props.clinicalInfo.familyHistory
              }
            </SummaryNotes>
          </Col>
        }
        {
          (this.props.clinicalInfo.attachments && this.props.clinicalInfo.attachments.length > 0) &&
          <Col md={12}>
            <SummaryTitle> Attachments </SummaryTitle>
              <div>
              { 
                this.props.clinicalInfo.attachments.map((attachment, $index) => {
                  return <FileList key={$index}>
                    <FileLink 
                      bsStyle="link" 
                      onClick={() => window.open(attachment.preview, '_blank')}
                      style={{ padding: '6px 0'}}
                    >
                      <Glyphicon glyph="file"/> {attachment.filename}
                    </FileLink>
                  </FileList>
                })
              }
            </div>
            <br />
          </Col>
        }

        <Col md={12}>
          <SummaryDetails> 
            <SummaryTitle> Consanguinity:  </SummaryTitle>
              {this.props.clinicalInfo.consanguinity ? 'Yes' : 'No'}
            </SummaryDetails>
        </Col>
        {
          (this.props.clinicalInfo.consanguinity && this.props.clinicalInfo.consanguinityInfo !== '') &&
          <Col md={12}>
            <SummaryTitle> Consanguinity Information </SummaryTitle>
            <SummaryNotes>
            {
              consanguinityNotes.length > 1 ?
              consanguinityNotes.map((ri, i) => {
                return <p key={i} style={{fontWeight:200}}>{ri}</p>;
              })
              : this.props.clinicalInfo.consanguinityInfo
            }
            </SummaryNotes>
          </Col>
        }
      </Row>
    </SummaryBox>
    )
  }
}

export default PatientDetailsModule;