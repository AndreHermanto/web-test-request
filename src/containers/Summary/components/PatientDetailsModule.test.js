import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import PatientDetailsModule from './PatientDetailsModule';

function getStep(value) {
  return value;
}
describe('PatientDetailsModule test', () => {
  const patientData = {
    "lastName":"abcabc",
    "firstName":"Ngabcuyen",
    "dob":"1-February-1917",
    "medicalRecordNo":"123123",
    "gender":"Male",
    "genderOther":"",
    "ethnicity":"Asian",
    "deceased":false,
    "sampleSource":"",
    "consent":false,
    "email":"abc@abc.gmail.com"
  }
  const clinicalData = {
    clinicalInfo: "Dissuade ecstatic and properly saw entirely sir why laughter endeavor. In on my jointure horrible margaret suitable he followed speedily. Indeed vanity excuse or mr lovers of on. By offer scale an stuff. Blush be sorry no sight. Sang lose of hour then he left find",
    consangunity: false,
    familyHistory: "kasdfbljkbfalhfasvfkjabskldfbkjsdfbkdashfvdjhfhb;kfjv;hfv;fjbafhb;akfd",
    relevantInvestigation:"asdkfhgasdlfhabsdlfjhalsfdjh",
    attachments:[{filename:'aaa', preview:null}]
  }
  const props = {
    patientDetails: patientData,
    clinicalInfo: clinicalData,
    handleOnClick: getStep
  };
    
  test('renders without crashing ', () => {
    const page = renderer.create(React.createElement(PatientDetailsModule, props)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <PatientDetailsModule patientDetails={patientData} clinicalInfo={clinicalData} handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step2')).toEqual('step2');
    expect(page.handleDetailsClick()).toEqual(2);
    expect(page.props.handleOnClick('step3')).toEqual('step3');
    expect(page.handleClinicalInfoClick()).toEqual(3);
  });
});
