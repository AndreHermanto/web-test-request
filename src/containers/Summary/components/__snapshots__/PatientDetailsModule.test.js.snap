exports[`PatientDetailsModule test renders without crashing  1`] = `
<div
  className="iKCvIz">
  <h3
    className="iArjKn">
     Patient - 
    Ngabcuyen abcabc
     
  </h3>
  <div
    className="dgEelt" />
  <div
    className="row">
    <div
      className="col-md-12">
      <h4
        className="hWtWAB">
        Details
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button">
          <span
            className="glyphicon glyphicon-pencil" />
           Edit
        </button>
      </h4>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Date of Birth: 
        </span>
        1917/February/1
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Medical record number:  
        </span>
        123123
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Gender:  
        </span>
        Male
      </p>
    </div>
    <div
      className="col-md-12">
      <h4
        className="hWtWAB">
        Clinical Information
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button">
          <span
            className="glyphicon glyphicon-pencil" />
           Edit
        </button>
      </h4>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Clinical note 
      </span>
      <div
        className="gczsYD">
        <p
          style={
            Object {
              "borderBottom": "1px solid #ccc",
              "fontWeight": 200,
              "paddingTop": 15,
            }
          } />
      </div>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Relevant investigation 
      </span>
      <div
        className="gczsYD">
        <p
          style={
            Object {
              "borderBottom": "1px solid #ccc",
              "fontWeight": 200,
            }
          }>
          asdkfhgasdlfhabsdlfjhalsfdjh
        </p>
      </div>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Family history 
      </span>
      <div
        className="gczsYD">
        <p
          style={
            Object {
              "borderBottom": "1px solid #ccc",
              "fontWeight": 200,
            }
          }>
          kasdfbljkbfalhfasvfkjabskldfbkjsdfbkdashfvdjhfhb;kfjv;hfv;fjbafhb;akfd
        </p>
      </div>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Attachments 
      </span>
      <div>
        <div
          className="iubaNK">
          <button
            className="jEoofW btn btn-link"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "padding": "6px 0",
              }
            }
            type="button">
            <span
              className="glyphicon glyphicon-file" />
             
            aaa
          </button>
        </div>
      </div>
      <br />
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Consanguinity:  
        </span>
        No
      </p>
    </div>
  </div>
</div>
`;
